UnityGLTF/Assets/UnityGLTF/Scripts/Extensions/GLTFSceneExporter.Skinning.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Collections.Generic;
+using GLTF.Schema;
+using UnityEngine;
+using System.Linq;
+
+namespace UnityGLTF
+{
+	public partial class GLTFSceneExporter
+	{
+		private static readonly Matrix4x4 InvertZMatrix = Matrix4x4.TRS(Vector3.zero, Quaternion.identity, new Vector3(1, 1, -1));
+		private static readonly Matrix4x4 InvertZMatrixInverse = InvertZMatrix.inverse;
+
+		//exported skins --> skinIds
+		private readonly Dictionary<Skin, SkinId> _skinCache = new Dictionary<Skin, SkinId>();
+
+		private void ExportSkins(IEnumerable<Transform> nodeTransforms)
+		{
+			foreach (var transform in nodeTransforms)
+			{
+				GameObject gameObject = transform.gameObject;
+				foreach (var skinnedMeshRenderer in gameObject.GetComponentsInChildren<SkinnedMeshRenderer>())
+				{
+					//grab the Unity node this skinned mesh renderer belongs to
+					var nodeId = this._nodeCache[skinnedMeshRenderer.transform];
+					//if: exist bind poses + a bone, OR exist bones
+					if (skinnedMeshRenderer.sharedMesh.bindposes != null && skinnedMeshRenderer.sharedMesh.bindposes.Any() &&
+						skinnedMeshRenderer.rootBone != null ||
+						skinnedMeshRenderer.bones != null && skinnedMeshRenderer.bones.Any())
+					{
+						//grab exorted node, and add to it its skin
+						_root.Nodes[nodeId.Id].Skin = ExportSkin(skinnedMeshRenderer);
+					}
+				}
+			}
+		}
+
+		private SkinId ExportSkin(SkinnedMeshRenderer skinnedMeshRenderer)
+		{
+			var skin = new Skin
+			{
+				BindPoses = skinnedMeshRenderer.sharedMesh.bindposes,
+				RootBone = skinnedMeshRenderer.rootBone,
+				Bones = skinnedMeshRenderer.bones,
+			};
+
+			SkinId skinId;
+			if (_skinCache.TryGetValue(skin, out skinId))
+			{
+				//already exported this skin, return it
+				return skinId;
+			}
+
+			skinId = new SkinId
+			{
+				Id = _root.Skins.Count,
+				Root = this._root
+			};
+
+			Transform parentTransform = skinnedMeshRenderer.GetComponentInParent<Transform>();
+			if (parentTransform != null && parentTransform.localScale != Vector3.one)
```

Comment:
This isn't enough. You need to check all the levels not just one level above. I also can't remember what happens if `skinnedMeshRenderer.transform` itself has scale.

Reply:
My bad, for some reason I had the idea that each node would check its parent anyways, which is not the case at all since not all nodes necessarily have a skin. Thank you!

Reply:
@bghgary I read this https://answers.unity.com/questions/147816/how-to-avoid-scaling-heritage-when-parenting.html, and it seems like the object is only affected by the direct parent's scale. Do we still need to propagate the warning to all parents (which might be costly)

Reply:
I don't think that thread has anything to do with this. You can probably get away with checking [lossyScale](https://docs.unity3d.com/ScriptReference/Transform-lossyScale.html).